__Creating the dataframe__

In [1]:
pip install beautifulsoup4

     |████████████████████████████████| 112kB 14.3MB/s eta 0:00:01
Note: you may need to restart the kernel to use updated packages.


In [2]:
pip install lxml

     |████████████████████████████████| 5.8MB 5.3MB/s eta 0:00:01
Note: you may need to restart the kernel to use updated packages.


In [4]:
from bs4 import BeautifulSoup
import pandas as pd
import requests
import lxml

In [5]:
source = requests.get('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M').text

In [6]:
soup = BeautifulSoup(source,'xml')

In [7]:
table = soup.find('tbody')


In [8]:
rows = []
trs = table.find_all('tr')
rows.append([td.get_text(strip = True) for td in trs[0].find_all('th')])
trs = trs[1:]
for tr in trs:
    rows.append([td.get_text(strip = True) for td in tr.find_all('td')])

In [9]:
df = pd.DataFrame(rows[1:], columns = rows[0])
df.head()

Postal Code           Borough               Neighborhood
0         M1A      Not assigned                           
1         M2A      Not assigned                           
2         M3A        North York                  Parkwoods
3         M4A        North York           Victoria Village
4         M5A  Downtown Toronto  Regent Park, Harbourfront

In [14]:
df_fil = df[df['Borough']!= 'Not assigned'].reset_index()
df_fil.shape

(103, 4)

In [15]:
df_fil.head()

index Postal Code           Borough  \
0      2         M3A        North York   
1      3         M4A        North York   
2      4         M5A  Downtown Toronto   
3      5         M6A        North York   
4      6         M7A  Downtown Toronto   

                                  Neighborhood  
0                                    Parkwoods  
1                             Victoria Village  
2                    Regent Park, Harbourfront  
3             Lawrence Manor, Lawrence Heights  
4  Queen's Park, Ontario Provincial Government

In [16]:
df_fil.drop(['index'],axis = 1, inplace = True)
df_fil.head()

Postal Code           Borough                                 Neighborhood
0         M3A        North York                                    Parkwoods
1         M4A        North York                             Victoria Village
2         M5A  Downtown Toronto                    Regent Park, Harbourfront
3         M6A        North York             Lawrence Manor, Lawrence Heights
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

In [17]:
df_fil['Postal Code'].unique

<bound method Series.unique of 0      M3A
1      M4A
2      M5A
3      M6A
4      M7A
      ... 
98     M8X
99     M4Y
100    M7Y
101    M8Y
102    M8Z
Name: Postal Code, Length: 103, dtype: object>

The Postal Code has same number of unique objects as number of rows of df_fil.

In [18]:
dfchk =df_fil[df_fil['Neighborhood']=='Not assigned']
dfchk.shape

(0, 3)

Neighborhood has no Not assigned rows

In [19]:
df_fil.head()

Postal Code           Borough                                 Neighborhood
0         M3A        North York                                    Parkwoods
1         M4A        North York                             Victoria Village
2         M5A  Downtown Toronto                    Regent Park, Harbourfront
3         M6A        North York             Lawrence Manor, Lawrence Heights
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

In [20]:
df_fil.shape

(103, 3)

df_fil is the required dataframe with 103 rows